sphinx_gallery/backreferences.py

```diff
@@ -258,7 +259,9 @@ def identify_names(script_blocks, ref_regex, global_variables=None, node=""):
     # Get matches from docstring inspection (explicit matches)
     text = "\n".join(block.content for block in script_blocks if block.type == "text")
     names.extend((x, x, False, False, True) for x in re.findall(ref_regex, text))
-    example_code_obj = dict()  # native dict preserves order nowadays
+    example_code_obj: dict[str, list[dict[str, Any]]] = (
```

Comment:
can be seen from code below:
```
        cobj = dict(
            name=attribute,
            module=module,
            module_short=module_short or module,
            is_class=is_class,
            is_explicit=is_explicit,
        )
        example_code_obj[name].append(cobj)
```
Actually, cobj would likely better be a namedtuple. But that's a change for a follow-up.